services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -866,6 +868,77 @@ class ProtectedPathSpec extends BaseSpecification {
+        given: "Two active neighboring switches with three paths at least"
+        def (Switch srcSwitch, Switch dstSwitch) = getNeighboringSwitchPair(3)
+
+        and: "A flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        and: "All alternative paths are unavailable (bring ports down on the source switch)"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        List<PathNode> broughtDownPorts = []
+        def altPaths = database.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path.findAll {
+            it != pathHelper.convert(northbound.getFlowPath(flow.id)) &&
+                    it != pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath)
+        }.unique { it.first() }
+
+        altPaths.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northbound.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        and: "ISL on a protected path is broken(bring port down) for changing the flow state to DOWN"
+        def protectedIslToBreak = pathHelper.getInvolvedIsls(currentProtectedPath)[0]
+        northbound.portDown(protectedIslToBreak.dstSwitch.dpId, protectedIslToBreak.dstPort)
+
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.DOWN }
+
+        when: "Make the current path less preferable than alternative path"
+        def alternativePath = altPaths.first()
+        def currentIsl = pathHelper.getInvolvedIsls(currentPath)[0]
+        def alternativeIsl = pathHelper.getInvolvedIsls(alternativePath)[0]
+
+        pathHelper.makePathMorePreferable(alternativePath, currentPath)
+
+        assert islUtils.getIslInfo(currentIsl).get().cost > islUtils.getIslInfo(alternativeIsl).get().cost
+
+        and: "Make alternative path available(bring port up on the source switch)"
+        northbound.portUp(alternativeIsl.srcSwitch.dpId, alternativeIsl.srcPort)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(alternativeIsl).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Flow state is changed to UP"
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Protected path is recalculated only"
+        def newFlowPathInfo = northbound.getFlowPath(flow.id)
+        pathHelper.convert(newFlowPathInfo) == currentPath
+        pathHelper.convert(newFlowPathInfo.protectedPath) != currentProtectedPath
```

Comment:
What about checking that `pathHelper.convert(newFlowPathInfo.protectedPath) == alternativePath` insted of this check?